File: CS/HideTotalsCustomCellValueExample/Form1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraPivotGrid;
using System.ComponentModel;
using System.Linq;

namespace HideTotalsCustomCellValueExample {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            pivotGridControl1.CustomCellValue += pivotGridControl1_CustomCellValue;
            BindingList<PivotData> data = new BindingList<PivotData>();
            for (int i = 0; i < 100; i++)
                data.Add(new PivotData() { Group = "Group " + i % 4, Row = "Row " + i % 5, 
                    Column = "Column" + i % 3, Value = i });
            pivotGridControl1.DataSource = data;
            pivotGridControl1.BestFitDataHeaders(true);
        }

        private void pivotGridControl1_CustomCellValue(object sender, PivotCellValueEventArgs e) {
            if (e.DataField == fieldLowLevel) {
                PivotGridField lastLevelField = 
                    pivotGridControl1.GetFieldsByArea(PivotArea.RowArea).Last();
                if(e.RowValueType == PivotGridValueType.Total 
                    || e.RowValueType == PivotGridValueType.GrandTotal 
                    || e.RowField != lastLevelField)
                    e.Value = null;
            } 
        }
    }
    class PivotData {
        public string Group { get; set; }
        public string Row { get; set; }
        public string Column { get; set; }
        public decimal Value { get; set; }
    }
}

File: VB/HideTotalsCustomCellValueExample/Form1.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraPivotGrid
Imports System.ComponentModel
Imports System.Linq

Namespace HideTotalsCustomCellValueExample
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			AddHandler pivotGridControl1.CustomCellValue, AddressOf pivotGridControl1_CustomCellValue
			Dim data As New BindingList(Of PivotData)()
			For i As Integer = 0 To 99
				data.Add(New PivotData() With {.Group = "Group " & i Mod 4, .Row = "Row " & i Mod 5, .Column = "Column" & i Mod 3, .Value = i})
			Next i
			pivotGridControl1.DataSource = data
			pivotGridControl1.BestFitDataHeaders(True)
		End Sub

		Private Sub pivotGridControl1_CustomCellValue(ByVal sender As Object, ByVal e As PivotCellValueEventArgs) Handles pivotGridControl1.CustomCellValue
			If e.DataField Is fieldLowLevel Then
				Dim lastLevelField As PivotGridField = pivotGridControl1.GetFieldsByArea(PivotArea.RowArea).Last()
				If e.RowValueType = PivotGridValueType.Total OrElse e.RowValueType = PivotGridValueType.GrandTotal OrElse e.RowField IsNot lastLevelField Then
					e.Value = Nothing
				End If
			End If
		End Sub

	End Class
	Friend Class PivotData
		Public Property Group() As String
		Public Property Row() As String
		Public Property Column() As String
		Public Property Value() As Decimal
	End Class
End Namespace
